### PR TITLE
Identify user with session

### PR DESCRIPTION
**Pull request by djasmine**
Enable session to identify user.
Add batch upload python script.

### CODE REVIEW

**Comment by yubowenok** [comment]
Please sync with the With Credentials APIs

File: server/server.js [hunk 1]
```diff
@@ -156,6 +157,46 @@ app.use(session({
 var exonFile = dataPath + 'exons.bin';
 
 /**
+ * User authenticate handler.
+ */
+app.use('/genotet', function(req, res, next) {
+
+  if (req.url.substr(0, 6) == '/check') {
```
Comment (yubowenok) [inline server/server.js]: You can use a separate get handler before this authentication handler?

```
app.get('/genotet/check', ...);
```

File: server/server.js [hunk 2]
```diff
@@ -156,6 +157,46 @@ app.use(session({
...
+    res.header('Access-Control-Allow-Origin', 'http://localhost');
+    res.jsonp({});
+  }
+  else if (req.url.substr(0, 5) == '/user') {
```
Comment (yubowenok) [inline server/server.js]: "else if" needs to be on the previous line.

File: server/server.js [hunk 3]
```diff
@@ -156,6 +157,46 @@ app.use(session({
...
+    next();
+  } else {
+    if (req.session.id === undefined) {
+      res.header('Access-Control-Allow-Origin', 'http://localhost');
```
Comment (yubowenok) [inline server/server.js]: Why not moving jsonResponse above and use that function instead?

File: server/server.js [hunk 4]
```diff
@@ -156,6 +157,46 @@ app.use(session({
...
+      log.serverLog('user not recognized.');
+      res.jsonp({
+        error: 'user not recognized.'
+      });
+    }
+    else {
```
Comment (yubowenok) [inline server/server.js]: else needs to be on the previous line.

File: server/server.js [hunk 5]
```diff
@@ -156,6 +157,46 @@ app.use(session({
...
+          log.serverLog('username', result);
+          req.username = result;
+          next();
+        } else {
+          log.serverLog(result.error);
+          if (result.error == 'no user info find.') {
+            req.username = 'anonymous';
+            next();
+          } else {
+            res.header('Access-Control-Allow-Origin', 'http://localhost');
```
Comment (yubowenok) [inline server/server.js]: You may use jsonResponse wrapper?

File: server/server.js [hunk 6]
```diff
@@ -179,6 +220,7 @@ var jsonpResponse = function(data, res) {
  */
 var jsonResponse = function(data, res) {
   // TODO(bowen): need to change cross-origin to https://localhost as well
+  res.header('Access-Control-Allow-Credentials', 'true');
```
Comment (yubowenok) [inline server/server.js]: Please sync.

File: server/server.js [hunk 7]
```diff
@@ -194,9 +236,10 @@ var jsonResponse = function(data, res) {
 /**
  * User log POST handler.
  */
-app.post('/genotet/log', function(req, res) {
+app.get('/genotet/log', function(req, res) {
```
Comment (yubowenok) [inline server/server.js]: Since now get/post both works with credentials, maybe we can just use post here.

File: server/server.js [hunk 8]
```diff
@@ -211,55 +254,58 @@ app.post('/genotet/upload', upload.single('file'), function(req, res) {
...
+  uploader.uploadFile(body, req.file, dataPath, bigWigToWigPath, req.username,
     function(ret) {
       jsonResponse(/** @type {Object} */(ret), res);
     });
 });
 
 
-app.post('/genotet/user', function(req, res) {
-  log.serverLog('POST user');
+app.get('/genotet/user', function(req, res) {
```
Comment (yubowenok) [inline server/server.js]: Since now get/post both works with credentials, maybe we can just keep POST here as well?

**Comment by djasmine** [comment]
You can review it again. @yubowenok

**Comment by yubowenok** [comment]
LGTM

**Pull request merged**